textpipe/doc.py

```diff
@@ -132,13 +154,29 @@ def clean_text(self, remove_html=True, clean_dots=True, clean_quotes=True,
     @property
     def ents(self):
         """
-        Extract a list of the named entities in text
+        A list of the named entities with sensible defaults.
 
         >>> doc = Doc('Sentence for testing Google text')
         >>> doc.ents
         [('Google', 'ORG')]
         """
-        return list({(ent.text, ent.label_) for ent in self.spacy_doc.ents})
+        return self.find_ents()
+
+    @hash_dict
+    @functools.lru_cache()
+    def find_ents(self, model_mapping=None):
```

Comment:
It makes more sense to me to pass a custom model or `model_name` here than a mapping, see also previous comment.

Reply:
I don't think this is necessary; the spacy_doc is used only internally (to compute other properties that the Doc object exposes). We could also decide to name `self.spacy_doc` `_spacy_doc` instead. @dodijk what do you think of this?

Reply:
I don't think it does; the model mapping is a lookup of `model_name` on language. If the text is in a different language, the doc should hence not use the custom model.

Reply:
My reasoning is that when you call `find_ents` on a doc the language is known (or knowable). The call `doc.find_ents(model_mapping[doc.language])` seems more logical from a `Doc` perspective than to specify a mapping in the call while the language is known. 

I can see how model mapping is useful in a `Pipeline` setting, but I would argue that the logic of model mapping would then move to there.

Reply:
Imagine this situation: you want to create a Doc from text, and use a different financials-specific NER-model that you trained separately for English and Dutch. I want to be able to initialize the doc with these models in order to get these financial entities in either Dutch or English and do not yet know the language (because the Doc will detect the language upon initialization).

This means that, in order to keep the Doc class a independently functioning unit that is able to use custom models, it should be able to pick the right model for a language on-the-spot.

Reply:
That is argued from a `Pipeline` perspective. Both models can be loaded on initialization and the call `doc.find_ents(model_mapping[doc.language])` would be just as efficient.

My objections to this mapping are twofold: 
1) I do not like the silently falling back on default model if the language is not in the mapping. That could yield in regular entities popping up in your financial entities example if the detected language was unexpected.
2) The additional complexity introduced in https://github.com/textpipe/textpipe/pull/52/files#diff-2888db35175db729bb748d941338d46cR203, https://github.com/textpipe/textpipe/pull/52/files#diff-2888db35175db729bb748d941338d46cR89 and https://github.com/textpipe/textpipe/pull/52/files#diff-845f7de7058e07e4399bc7ca520fe68fR8, that seems unnecessary to me for the functionality that we need.

1 can be resolved in other ways, and in fact, I am not entirely sure what happens now if we detect a language that SpaCy does not support and then ask for entities.

Reply:
I strongly agree on point 1. Silent falling back is something to always be avoided. I prefer exceptions rather than silent fallbacks.

Having said that, I am not certain what an alternative would be. In the case of @dodijk 's proposal of doc.find_ents(model_mapping[doc.language]) , I would suggest the argument passed is a list and not a single mapping.